core/src/org/javarosa/xpath/expr/XPathPathExpr.java

```diff
@@ -170,7 +177,8 @@ public TreeReference getReference(boolean allowPredicates) throws XPathUnsupport
                 for (int j = 0; j < step.predicates.length; j++) {
                     v.addElement(step.predicates[j]);
                 }
-                ref.addPredicate(i, v);
```

Comment:
If a (nested) predicate has ../, then "i" doesn't represent the correct step position to add the predicate to. This causes crashes on expressions like:
/data/group[count(../team[@id = 'mobile']) > 0])